sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update_slm.hpp

```diff
@@ -0,0 +1,720 @@
+constexpr int64_t n_groups = 1;
+constexpr int64_t start_ind = 3;
+constexpr int64_t masked_lane = 1;
+constexpr int64_t repeat = 1;
+constexpr int64_t stride = 4;
+
+// Helper functions
+
+template <class, int, template <class, int> class> class TestID;
+
+const char *to_string(atomic_op op) {
+  switch (op) {
+  case atomic_op::add:
+    return "add";
+  case atomic_op::sub:
+    return "sub";
+  case atomic_op::inc:
+    return "inc";
+  case atomic_op::dec:
+    return "dec";
+  case atomic_op::umin:
+    return "umin";
+  case atomic_op::umax:
+    return "umax";
+  case atomic_op::xchg:
+    return "xchg";
+  case atomic_op::cmpxchg:
+    return "cmpxchg";
+  case atomic_op::bit_and:
+    return "bit_and";
+  case atomic_op::bit_or:
+    return "bit_or";
+  case atomic_op::bit_xor:
+    return "bit_xor";
+  case atomic_op::smin:
+    return "smin";
+  case atomic_op::smax:
+    return "smax";
+  case atomic_op::fmax:
+    return "fmax";
+  case atomic_op::fmin:
+    return "fmin";
+  case atomic_op::fadd:
+    return "fadd";
+  case atomic_op::fsub:
+    return "fsub";
+  case atomic_op::fcmpxchg:
+    return "fcmpxchg";
+  case atomic_op::load:
+    return "load";
+  case atomic_op::store:
+    return "store";
+  case atomic_op::predec:
+    return "predec";
+  }
+  return "<unknown>";
+}
+
+template <int N> inline bool any(simd_mask<N> m, simd_mask<N> ignore_mask) {
+  simd_mask<N> m1 = 0;
+  m.merge(m1, ignore_mask);
+  return m.any();
+}
+
+// The main test function
+
+template <class T, int N, template <class, int> class ImplF, bool UseMask>
+bool test_slm(queue q) {
+  constexpr auto op = ImplF<T, N>::atomic_op;
+  using CurAtomicOpT = decltype(op);
+  constexpr int n_args = ImplF<T, N>::n_args;
+
+  std::cout << "SLM testing" << " op=" << to_string(op)
+            << " T=" << esimd_test::type_name<T>() << " N=" << N << "\n\t"
+            << " UseMask=" << (UseMask ? "true" : "false")
+            << "{ thr_per_group=" << threads_per_group
+            << " n_groups=" << n_groups << " start_ind=" << start_ind
+            << " masked_lane=" << masked_lane << " repeat=" << repeat
+            << " stride=" << stride << " }...";
+
+  constexpr size_t size = start_ind + (N - 1) * stride + 1;
+  T *arr = malloc_shared<T>(size, q);
+  constexpr int n_threads = threads_per_group * n_groups;
+
+  for (int i = 0; i < size; ++i) {
+    arr[i] = ImplF<T, N>::init(i);
+  }
+
+  range<1> glob_rng(n_threads);
+  range<1> loc_rng(threads_per_group);
+  nd_range<1> rng(glob_rng, loc_rng);
+
+  try {
+    auto e = q.submit([&](handler &cgh) {
+      cgh.parallel_for(rng, [=](id<1> ii) SYCL_ESIMD_KERNEL {
+        int i = ii;
+        slm_init<32768>();
+        simd<uint32_t, N> offsets(start_ind * sizeof(T), stride * sizeof(T));
+        simd<T, size> data;
+        data.copy_from(arr);
+
+        simd<uint32_t, size> slm_offsets(0, sizeof(T));
+        // TODO: replace this API with unified version.
+        lsc_slm_scatter(slm_offsets, data);
```

Comment:
slm_offsets here are: 0, sizeof(T), 2*sizeof(T), ...
Why not to use slm_block_store() ?

Reply:
Done.